core/src/verifier.rs

```diff
@@ -146,91 +160,164 @@ where
                 kickoff_utxo.outpoint.vout.to_be_bytes()
             );
 
+            // Check if they are really the operators that sent these kickoff_utxos
             utils::SECP.verify_schnorr(
                 &operators_kickoff_sigs[i],
                 &secp256k1::Message::from_digest(kickoff_sig_hash),
                 &self.config.operators_xonly_pks[i],
             )?;
+
+            // Check if for each operator the address of the kickoff_utxo is correct TODO: Maybe handle the possible errors better
+            let musig2_and_operator_script =
+                script_builder::create_musig2_and_operator_multisig_script(
+                    &self.nofn_xonly_pk,
+                    &self.config.operators_xonly_pks[i],
+                );
+            let (musig2_and_operator_address, _) = TransactionBuilder::create_taproot_address(
```

Comment:
Lets have a create_kickoff_address function in transaction_builder

Reply:
Done